### PR TITLE
EFS > Update creation and storage class

### PR DESCRIPTION
**Pull request by jana-opszero**
Using this method https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/examples/kubernetes/dynamic_provisioning/README.md

### CODE REVIEW

File: efs.tf [hunk 1]
```diff
@@ -1,29 +1,84 @@
...
   storage_provisioner = "efs.csi.aws.com"
-  parameters = {
-    provisioningMode = "efs-ap"
-    fileSystemId     = aws_efs_file_system.this.id
-    directoryPerms   = "777"
-    gidRangeStart    = "1000"    # optional
-    gidRangeEnd      = "7000000" # optional
+}
+
+resource "kubernetes_persistent_volume" "efs_pvc" {
```
Comment (jana-opszero) [inline efs.tf]: We can move `kubernetes_persistent_volume` and `kubernetes_persistent_volume_claim` to customer side

File: efs.tf [hunk 2]
```diff
@@ -1,29 +1,84 @@
-resource "aws_efs_file_system" "this" {
+resource "aws_efs_file_system" "file_system" {
```
Comment (abhiyerra) [inline efs.tf]: `this`

File: efs.tf [hunk 3]
```diff
@@ -1,29 +1,84 @@
...
+  tags = {
+    Name        = "${var.name}-eks-efs"
+    Environment = var.name
+  }
 }
 
-resource "aws_efs_mount_target" "this" {
-  count           = length(var.subnet_ids)
-  file_system_id  = aws_efs_file_system.this.id
+resource "aws_efs_mount_target" "mount_target" {
```
Comment (abhiyerra) [inline efs.tf]:
```
this
```

**Review by abhiyerra — CHANGES_REQUESTED** [review]

File: efs.tf [hunk 4]
```diff
@@ -1,29 +1,84 @@
...
   storage_provisioner = "efs.csi.aws.com"
-  parameters = {
-    provisioningMode = "efs-ap"
-    fileSystemId     = aws_efs_file_system.this.id
-    directoryPerms   = "777"
-    gidRangeStart    = "1000"    # optional
-    gidRangeEnd      = "7000000" # optional
+}
+
+resource "kubernetes_persistent_volume" "efs_pvc" {
```
Comment (abhiyerra) [inline efs.tf]: Name `this`

File: efs.tf [hunk 5]
```diff
@@ -1,29 +1,84 @@
...
+    storage_class_name               = kubernetes_storage_class.efs_sc.metadata[0].name
+
+    csi {
+      driver        = "efs.csi.aws.com"
+      volume_handle = aws_efs_file_system.file_system.id
+    }
+  }
+}
+
+resource "kubernetes_persistent_volume_claim" "efs_storage_claim" {
```
Comment (abhiyerra) [inline efs.tf]: Name `this`

**Review by abhiyerra — CHANGES_REQUESTED** [review]

**Pull request merged**